apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/coverage/model/BeneficiaryCoverage.java

```diff
@@ -295,6 +339,74 @@ private Coverage mapCoverageDual(Coverage coverage, ProfileType profileType, Str
     return coverage;
   }
 
+  private Coverage mapCoverageC(
+      Coverage coverage, CoveragePart coveragePart, ProfileType profileType, String orgId) {
+    var enrollmentOpt = getEnrollment(coveragePart);
+
+    identifier.toFhir(orgId).ifPresent(coverage::addIdentifier);
+
+    return enrollmentOpt
+        .map(
+            enrollment -> mapBaseCoverageCD(coverage, coveragePart, enrollment, profileType, orgId))
+        .orElseGet(() -> toEmptyResource(coverage));
+  }
+
+  private Coverage mapCoverageD(
+      Coverage coverage, CoveragePart coveragePart, ProfileType profileType, String orgId) {
+    var enrollmentOpt = getEnrollment(coveragePart);
+    if (enrollmentOpt.isEmpty()) {
+      return toEmptyResource(coverage);
+    }
+
+    mapBaseCoverageCD(coverage, coveragePart, enrollmentOpt.get(), profileType, orgId);
```

Comment:
I know it's not strictly necessary, but it's nice to utilize the return value here to make it clearer to readers that `coverage` is being modified here.
```suggestion
    coverage = mapBaseCoverageCD(coverage, coveragePart, enrollmentOpt.get(), profileType, orgId);
```